src/validation.cpp

```diff
@@ -2868,7 +2843,7 @@ bool CChainState::ActivateBestChain(BlockValidationState &state, const CChainPar
                 }
                 pindexNewTip = m_chain.Tip();
 
-                for (const PerBlockConnectTrace& trace : connectTrace.GetBlocksConnected()) {
+                for (auto& trace : connectTrace) {
```

Comment:
Why is this change to auto? `auto` shouldn't be used in consensus code unless there is a reason

Reply:
Sounds sensible. I'll revert.